sched/sched/sched_addreadytorun.c

```diff
@@ -158,6 +169,17 @@ bool nxsched_add_readytorun(FAR struct tcb_s *btcb)
   int cpu;
   int me;
 
+#if CONFIG_RR_INTERVAL > 0
+#ifdef CONFIG_SCHED_SPORADIC
+  if ((btcb->flags & TCB_FLAG_POLICY_MASK) == TCB_FLAG_SCHED_RR)
+#endif
+    {
+      /* Reset the task's timeslice. */
+
+      btcb->timeslice = MSEC2TICK(CONFIG_RR_INTERVAL);
```

Comment:
why need reset the time slice in nxsched_add_readytorun, I think it's enough to simply remove the reset in nxsched_resume_scheduler.

Reply:
When a task is added to ready queue, the task is at the end of the same priority. When the task runs again, a long time has passed. So we re-assign the longest-timeslice to the task. @xiaoxiang781216

Reply:
But nxsched_process_roundrobin will reset time slice for this case:
https://github.com/apache/incubator-nuttx/blob/master/sched/sched/sched_roundrobin.c#L102-L130

Reply:
That's after the timeslice runs out, the task will be switch out immediately. Tasks that actively give up the cpu should be rewarded.

Reply:
But from https://man7.org/linux/man-pages/man7/sched.7.html:
A SCHED_RR thread that has been preempted by a higher priority thread and subsequently resumes execution as a running thread will complete the unexpired portion of its round-robin time quantum.
At least, Linux kernel doesn't reset the time slice after suspend/resume cycle.

Reply:
For the first point, after applying this commit, the behavior is consistent with linux kernel.
For the second one, the scheduler works too if doesn't reset the task‘s timeslice, of course.
But I think it would be better to reset the task's timeslice.

Reply:
Yes, but the standard compliance is the highest priority of NuttX.

Reply:
@zhaoxiu-zeng what's about your thinking?

Reply:
I don't think this commit undermines posix compliance. This is what most OS do, for example ThreadX. I will try to do a commit to linux kernel.

Reply:
could you share the email thread after post the patch? So we can know the thinking from Linux community.

Reply:
https://lkml.org/lkml/2021/12/22/188

Reply:
Nice, let's watch the feedback.

Reply:
@zhaoxiu-zeng look like Linux community also think the change doesn't conform to POSIX standard.

Reply:
Do you think we should remove the change in nxsched_add_readytorun?

Reply:
We can make such a modification first. The question of whether to reset the timeslice could be left for future.

Reply:
Ok, please update your PR, so I can merge it once it pass the CI.